pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -31,22 +32,47 @@ public class BufferedData implements BufferedSequentialData, ReadableSequentialD
     /** {@link ByteBuffer} used as backing buffer for this instance */
     private final ByteBuffer buffer;
 
+    private final boolean direct;
+
     /**
      * Wrap an existing allocated {@link ByteBuffer}. No copy is made.
      *
      * @param buffer the {@link ByteBuffer} to wrap
+     * @param direct If this is a dirrect buffer allocation.
      */
-    private BufferedData(@NonNull final ByteBuffer buffer) {
+    private BufferedData(@NonNull final ByteBuffer buffer, final boolean direct) {
```

Comment:
Think you can remove direct parameter as you can just call `ByteBuffer.isDirect()`